packages/expo-modules-core/android/src/main/cpp/JavaScriptObject.cpp

```diff
@@ -36,23 +36,28 @@ JavaScriptObject::JavaScriptObject(
   std::weak_ptr<JavaScriptRuntime> runtime,
   std::shared_ptr<jsi::Object> jsObject
 ) : runtimeHolder(std::move(runtime)), jsObject(std::move(jsObject)) {
-  assert(runtimeHolder.lock() != nullptr);
+  runtimeHolder.ensureRuntimeIsValid();
+}
+
+JavaScriptObject::JavaScriptObject(
+  WeakRuntimeHolder runtime,
+  std::shared_ptr<jsi::Object> jsObject
+) : runtimeHolder(std::move(runtime)), jsObject(std::move(jsObject)) {
+  runtimeHolder.ensureRuntimeIsValid();
```

Comment:
If `WeakRuntimeHolder` inherits from `std::weak_ptr<JavaScriptRuntime>`, why do we need this another constructor? 🤔

Reply:
It reduces the number of construction calls. Without that when you try to construct `JavaScriptObject` from `WeakRuntimeHolder` it will also call the `WeakRuntimeHolder` contractor from ptr.